src/Selenium2Library/keywords/_element.py

```diff
@@ -297,6 +297,21 @@ def get_horizontal_position(self, locator):
             raise AssertionError("Could not determine position for '%s'" % (locator))
         return element.location['x']
 
+    def get_element_size(self, locator):
+        """Returns width and height of element identified by `locator`.
+
+        The element width and height is returned.
+        Fails if a matching element is not found.
+        """
+        element = self._element_find(locator, True, False)
+        if element is None:
+            raise AssertionError("Element with locator '%s' was not found or is not visible" % (locator))
+
+        if element.size['width'] * element.size['height'] == 0:
```

Comment:
this is was a request I've been working on. previous version did not raise
any exception and pull request has been rejected.

On Thu, Nov 5, 2015 at 8:43 PM, Tatu Aalto <notifications@github.com> wrote:

> In src/Selenium2Library/keywords/_element.py
> <https://github.com/robotframework/Selenium2Library/pull/528#discussion_r44050715>
> :
>
> > @@ -297,6 +297,21 @@ def get_horizontal_position(self, locator):
> >              raise AssertionError("Could not determine position for '%s'" % (locator))
> >          return element.location['x']
> >
> > +    def get_element_size(self, locator):
> > +        """Returns width and height of element identified by `locator`.
> > +
> > +        The element width and height is returned.
> > +        Fails if a matching element is not found.
> > +        """
> > +        element = self._element_find(locator, True, False)
> > +        if element is None:
> > +            raise AssertionError("Element with locator '%s' was not found or is not visible" % (locator))
> > +
> > +        if element.size['width'] * element.size['height'] == 0:
>
> I am not sure, is it a good idea to raise an error if size is zero?
> Selenium seems to return zero size, if the element is hidden and I am not
> totally sure is it a good idea to change that idea. But raising an error
> follows Selenium2Library general idea. So we need discuss which way to go.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/robotframework/Selenium2Library/pull/528/files#r44050715>
> .
>